### DISCUSSION TITLE
[Question]: LLM on Ollama endpoints is not responding based on a document imported from the RAG API.

### DISCUSSION THREAD

**Discussion by SEOLJINYOUNG** [danny-avila/LibreChat#3920] [Troubleshooting]
### What is your question?

LLM on Ollama endpoints is not responding based on a document imported from the RAG API.

EMBEDDINGS_PROVIDER=ollama
EMBEDDINGS_MODEL=nomic-embed-text
MODEL=llama3.1:70b

I took a log app\clients\OllamaClient.js, 
and on the messages list
It contains two messages.
messages = [
{role: 'system', content:'content from rag api'},
{role: 'user', content:'user input'},
]

### More Details

When I ask you to tell me the contents of the attached document, llm's answer says that you don't save or remember the previous conversation, or answers based on existing pre-learned content.

I'd appreciate it if you could give me some ideas on which parts to look at.

### What is the main subject of your question?

Endpoints, User System/OAuth, Other

### Screenshots

![image](https://github.com/user-attachments/assets/8b699486-f99a-41ae-a392-d2eef44c9aba)
![image](https://github.com/user-attachments/assets/83e7e57d-48c1-4f85-a7be-796884d34b5a)
![image](https://github.com/user-attachments/assets/eff12eba-b3f6-40d3-9bd9-c7ba86032fa8)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lokdao** [comment]
Hi, I am facing a similar issue.
I am serving llama3.2 via Ollama and am also using the nomic-embed-text through Ollama.
When uploading a document, llama3.2 does not detect it. However using ChatGPT-4o via Azure Open AI with the same file works perfectly fine.

### Config
**docker-compose-override.yaml**
<img width="482" alt="Screenshot 2024-10-01 at 00 44 34" src="https://github.com/user-attachments/assets/01f158c0-2d4c-41c3-b69b-e7ca990a122e">

**.env**
<img width="487" alt="Screenshot 2024-10-01 at 00 44 06" src="https://github.com/user-attachments/assets/76833ec1-77af-4744-8c06-da9fe6b263f0">

### Prompt
<img width="776" alt="Screenshot 2024-10-01 at 00 46 03" src="https://github.com/user-attachments/assets/dad87eae-2b5d-4022-a7f6-b73dfdd62633">

**Comment by aleibovici** [comment]
I have the exact same issue with a Ollama and uploaded documents. 
If I change from Ollama to OpenAI, I receive an adequate response.
That tells me that the RAG API with nomic-embedding is working.

It will only work if I force-prompt, but that's not a solution.

**Comment by parthpat12** [comment]
I’m having same issue. Is there any workaround or fix?

**Comment by SEOLJINYOUNG** [comment]
@parthpat12 
I changed the model and modified the prompts that configure the document content.
llama3->gemma2

Try changing the model first without modifying the prompt.

**Reply by parthpat12** [reply]
Changing model did work. Thanks

**Comment by litznerski** [comment]
Same problem here :-( It is impossible to get rid of openAPI when using RAG_API - impossible to use Ollama for Embeddings. A Perplexity Deep Research shows, that this is a common issue - seems to be hardcoded somewhere.
